recipes/lua/all/CMakeLists.txt

```
@@ -22,37 +17,27 @@ conan_basic_setup()
 #     - This is a source-incompatible change because extern "C" is chosen by the including application.
 # - The lua.hpp header will not be available.
 
-PROJECT ( lua )
-
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8)
-
 SET(SOURCE_DIR ${CMAKE_SOURCE_DIR}/${SOURCE_SUBDIR} )
 
-SET (HDR_LIBLUA
-${SOURCE_DIR}/src/lapi.h ${SOURCE_DIR}/src/lauxlib.h ${SOURCE_DIR}/src/lcode.h ${SOURCE_DIR}/src/lctype.h ${SOURCE_DIR}/src/ldebug.h ${SOURCE_DIR}/src/ldo.h ${SOURCE_DIR}/src/lfunc.h
-${SOURCE_DIR}/src/lgc.h ${SOURCE_DIR}/src/llex.h ${SOURCE_DIR}/src/llimits.h ${SOURCE_DIR}/src/lmem.h ${SOURCE_DIR}/src/lobject.h ${SOURCE_DIR}/src/lopcodes.h ${SOURCE_DIR}/src/lparser.h
-${SOURCE_DIR}/src/lstate.h ${SOURCE_DIR}/src/lstring.h ${SOURCE_DIR}/src/ltable.h ${SOURCE_DIR}/src/ltm.h ${SOURCE_DIR}/src/lua.h ${SOURCE_DIR}/src/luaconf.h ${SOURCE_DIR}/src/lualib.h
-${SOURCE_DIR}/src/lundump.h ${SOURCE_DIR}/src/lvm.h ${SOURCE_DIR}/src/lzio.h
-)
-
 # Build Libraries
 SET (SRC_LIBLUA
-${SOURCE_DIR}/src/lapi.c ${SOURCE_DIR}/src/lauxlib.c ${SOURCE_DIR}/src/lbaselib.c ${SOURCE_DIR}/src/lbitlib.c ${SOURCE_DIR}/src/lcode.c ${SOURCE_DIR}/src/lcorolib.c
+${SOURCE_DIR}/src/lapi.c ${SOURCE_DIR}/src/lauxlib.c ${SOURCE_DIR}/src/lbaselib.c ${SOURCE_DIR}/src/lcode.c ${SOURCE_DIR}/src/lcorolib.c
 ${SOURCE_DIR}/src/lctype.c ${SOURCE_DIR}/src/ldblib.c ${SOURCE_DIR}/src/ldebug.c ${SOURCE_DIR}/src/ldo.c ${SOURCE_DIR}/src/ldump.c ${SOURCE_DIR}/src/lfunc.c ${SOURCE_DIR}/src/lgc.c
 ${SOURCE_DIR}/src/linit.c ${SOURCE_DIR}/src/liolib.c ${SOURCE_DIR}/src/llex.c ${SOURCE_DIR}/src/lmathlib.c ${SOURCE_DIR}/src/lmem.c ${SOURCE_DIR}/src/loadlib.c ${SOURCE_DIR}/src/lobject.c
 ${SOURCE_DIR}/src/lopcodes.c ${SOURCE_DIR}/src/loslib.c ${SOURCE_DIR}/src/lparser.c ${SOURCE_DIR}/src/lstate.c ${SOURCE_DIR}/src/lstring.c ${SOURCE_DIR}/src/lstrlib.c
 ${SOURCE_DIR}/src/ltable.c ${SOURCE_DIR}/src/ltablib.c ${SOURCE_DIR}/src/ltm.c ${SOURCE_DIR}/src/lundump.c ${SOURCE_DIR}/src/lutf8lib.c ${SOURCE_DIR}/src/lvm.c ${SOURCE_DIR}/src/lzio.c
 )
 
+IF(LUA_VERSION VERSION_LESS 5.4.0)
+    LIST(APPEND SRC_LIBLUA ${SOURCE_DIR}/src/lbitlib.c)
+ENDIF()
```

Would this be *simpler* and more future proof without needing any special casing of versions?
```suggestion
file(GLOB SRC_LIBLUA "${SOURCE_DIR}/src/*.c")
list(REMOVE_ITEM SRC_LIBLUA "${SOURCE_DIR}/src/lua.c" "${SOURCE_DIR}/src/luac.c")
```

---

Good point, done.